futures-util/src/stream/mod.rs

```diff
@@ -868,6 +868,9 @@ pub trait StreamExt: Stream {
     /// This combinator will attempt to pull items from both streams. Each
     /// stream will be polled in a round-robin fashion, and whenever a stream is
     /// ready to yield an item that item is yielded.
+    ///
+    /// When either this stream or the provided one ends, the remaining stream will be polled to
+    /// completion. The stream that ended will no longer be polled.
```

Comment:
I'd phrase it like this:

> After the first input stream completes, the remaining one will be polled exclusively. The returned stream completes when both input stream have completed.

(Note: 80 char limit)

Feel free to tweak it further if you want ;)

Reply:
`
/// After the consumed stream or the argument stream ends, the remaining 
    /// stream will be polled exclusively. The stream returned from select will
    /// complete when both the consumed stream and the argument stream have 
    /// completed.`

First input stream reads a little weird to me, possibly indicating the consumed stream.

Not sure what the correct terminology is for extension/enrichment traits like this, where one argument is self which is moved/consumed by the function, and the other argument is provided.

Maybe actually tackle the whole documentation with something like `
Select is a combinator which consumes this stream and the stream provided as an argument, producing a pair of input streams. These input streams are polled in round-robin fashion, and whenever a stream is ready to yield an item, that item is yielded. 

When an input stream completes, the remaining input stream from the pair is polled exclusively. The stream returned by the Select combinator will complete when both input streams from the pair have completed.`

Reply:
How about: 
> After one of the two input stream completes, the remaining one will be polled exclusively. The returned stream completes when both input stream have completed.

I don't want to emphasize too much that one stream is the `self` argument and the other the following argument. It doesn't really matter in this context. Also, it's very like that we soon make `join` and `select` free functions. `select(stream1, stream2)` is much more intuitive `stream1.select(stream2)`

As for the other thing you mention, various future combinators mentions this:

> Note that this function consumes this future and returns a wrapped version of it.

This can be adapted for this combinator. Of course use developer can glean the same by looking at the function signature.

Reply:
```
This combinator will attempt to pull items from both streams. Each
stream will be polled in a round-robin fashion, and whenever a stream is
ready to yield an item that item is yielded.

After one of the two input stream completes, the remaining one will be 
polled exclusively. The returned stream completes when both input 
streams have completed.

Note that this function consumes this stream and returns a wrapped version of it.```

That work?